### DISCUSSION TITLE
Connect to AWS Aurora Postgres using rds iam token

### DISCUSSION THREAD

**Discussion by ssalihu** [prisma/prisma#20099] [Q&A] [answered]
I am trying to connect to AWS Aurora Postgres using the RDS IAM token as the password. The RDS token generated is a long string with special characters in it. Here is the error that i get when using iam token

`The provided database string is invalid. Error parsing connection string: invalid port number in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.`

```js
dbAuthToken = getDBAuthToken()
url = `postgresql://user01:${dbAuthToken}@aws-dev-rds-prjdbregionalclustera.cluster-cxzkasvsmucey.us-east-1.rds.amazonaws.com:5432/testdb?connection_limit=1`

  prisma = new PrismaClient({
    datasources: {db:{url}}
  });    
  output = await prisma.timestampTable.findMany({where:{'ts_val':'1111'}})
```
```js
const { Signer } = require("@aws-sdk/rds-signer");
 
const signer = new Signer({
    region: 'us-east-1',
    hostname: 'aws-dev-rds-prjdbregionalclustera.cluster-cxzkasvsmucey.us-east-1.rds.amazonaws.com',
    port: 5432,
    username: 'user01'     
})
async function getDBAuthToken () {
    let dbAuthToken  = null;
    try {
        dbAuthToken = await signer.getAuthToken();
          console.log('Fetched dbAuthToken :', dbAuthToken)
          return dbAuthToken;
    } catch(err) {
        console.log(`Failed to get auth token: ${err}`);
        return null;
    }
}
```

Has anyone used AWS RDS IAM token as the password in Prisma Client?

**Comment by nurul3101** [comment, marked as answer]
Hello @ssalihu 👋 

The problem you're experiencing is likely due to the special characters within the RDS IAM token. The token can include characters that are not valid within a URL.

Can you try passing the value of `dbAuthToken` to [encodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) before passing it in the database URL? This should ensure that the RDS IAM token is correctly encoded before being included within the URL.

Please feel free to **mark this Discussion as answered** in case this fixes the issue. Also, please let us know in case this doesn't fix it, so that we can try any other alternative solutions.

**Reply by nurul3101** [reply]
Just checking to see if you had a chance to look at my last response. Please feel free to let us know in case you need any assistance.

**Reply by ssalihu** [reply]
Thank you for the followup. I used `encodedURIComponent` and it works now!

**Reply by nurul3101** [reply]
I am glad to hear that! 🙂
